src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/oracle/OracleEngineSchemaTest.java

```diff
@@ -100,4 +106,68 @@ public void testSystemGeneratedColumns() throws Exception {
             engine.close();
         }
     }
+
+    /**
+     * After changing the oracle double data type from DOUBLE PRECISION to BINARY_DOUBLE the special
+     * value 'NaN' should be inserted into the database without any error.
+     */
+    @Test
+    public void testInsertNan() throws Exception {
+        testInsertSpecialValues("NaN");
+    }
+
+    /**
+     * After changing the oracle double data type from DOUBLE PRECISION to BINARY_DOUBLE the special
+     * value 'Infinity' should be inserted into the database without any error.
+     */
+    @Test
+    public void testInsertInfinity() throws Exception {
+        testInsertSpecialValues("Infinity");
+    }
+
+    /**
+     * The 'randomString' is not a special value for the BINARY_DOUBLE type so it should throw an error.
+     */
+    @Test
+    public void testRandomValuesDoNoWorkInBinaryDoubleColumn() throws Exception {
+        try {
+            testInsertSpecialValues("randomString");
+        } catch (DatabaseEngineException e) {
+            // It is supposed to.
+            return;
+        }
+
+        fail("Should have thrown an exception");
+    }
+
+    /**
+     * Auxiliary method to insert a provided special value for the BINARY_DOUBLE oracle datatype.
+     *
+     * @param columnValue The column value.
+     * @throws Exception If something goes wrong storing data into the database.
+     */
+    private void testInsertSpecialValues(final String columnValue) throws Exception {
+        final DatabaseEngine engine = DatabaseFactory.getConnection(properties);
+        try {
+            final DbEntity entity = dbEntity()
+                    .name("TEST_DOUBLE_COLUMN")
+                    .addColumn("id", INT)
+                    .addColumn("DBL", DOUBLE)
+                    .pkFields("id")
+                    .build();
+            engine.addEntity(entity);
+
+            final EntityEntry entry = entry()
+                    .set("id", 1)
+                    .set("DBL", columnValue)
+                    .build();
+
+            engine.persist(entity.getName(), entry);
+            final List<Map<String, ResultColumn>> dbl = engine.query(select(column("DBL")).from(table(entity.getName())));
```

Comment:
This is only one way of inserting the value, you can see in LongOverflowTest examples of other ways to do it that are not covered in this test.